src/bootstrap/cache.rs

```diff
@@ -169,19 +169,21 @@ impl Ord for Interned<String> {
     }
 }
 
-struct TyIntern<T> {
+struct TyIntern<T: Hash + Clone + Eq> {
     items: Vec<T>,
     set: HashMap<T, Interned<T>>,
 }
 
-impl<T: Hash + Clone + Eq> TyIntern<T> {
-    fn new() -> TyIntern<T> {
+impl<T: Hash + Clone + Eq> Default for TyIntern<T> {
```

Comment:
Can this be derived?

Reply:
No; the derive `impl` would have an additional `where T: Default`.

A comment saying so might be a good idea though.

Reply:
Get it